binary_search/binary_search.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *
+ * @author jsroyal
+ */
+public class BinarySearch {
+
+    public static void main(String[] args) {
+        int arr[] = {1, 5, 35, 112, 258, 324},
+                search_arr[] = {1, 35, 112, 324, 67},
+                pos, i;
+        BinarySearch obj = new BinarySearch();
```

Comment:
This is true that we won't be `OOP`ing much if we go with the sample code. But that's not needed with problems of this type. As it was already discussed at many places that we have to keep the code as function/method based as possible due to the fact that these are just implementation of mathematical algorithms.

We may use these concepts in other programs where `OO` design helps simplify the problem. But that is not the case here.

I, therefore, request you to go with the sample code for the case.